assets/src/utils/parse-infinity.js

```diff
@@ -20,13 +20,10 @@ const parseInfinity = ( number, asInt = false, forDb = false ) => {
 		value === 'INF' ||
 		value === Infinity ||
 		isNil( value );
-	number = representsInfinity( number ) || (
-		number.type &&
-		number.type.name === 'InfinitySymbol' &&
-		representsInfinity( number.props.value )
-	) ?
-		Infinity :
-		number;
+	if ( number && number.type && number.type.name === 'InfinitySymbol' ) {
```

Comment:
It would be nice to have this plugin: https://babeljs.io/docs/en/next/babel-plugin-proposal-optional-chaining.html
Have used it on previous project, didn't had any issue.
With that plugin, we'd have:
`	if ( numbe?r.type?.name && number.type.name === 'InfinitySymbol' ) { ...`
***
note: this is optional, thus I do approve the PR with the current solution

Reply:
Yes, optional chaining can be enabled. I have some other suggestions for that part of the code that I want to discuss with you and @tn3rb